test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/RequestTests.cs

```diff
@@ -184,6 +187,51 @@ public Task RemoteIPv6Address()
             }
         }
 
+        [Fact]
+        public async Task ConnectionResetAbortsRequest()
+        {
+            const int connectionErrorEventId = 14; // from KestrelTrace.cs
+
+            var testSink = new TestSink(write => write.EventId.Id == connectionErrorEventId);
+            var requestAbortedAcquired = new CancellationTokenSource();
+            var builder = new WebHostBuilder()
+                .UseLoggerFactory(new TestLoggerFactory(testSink, true))
+                .UseKestrel()
+                .UseUrls($"http://127.0.0.1:0")
+                .Configure(app => app.Run(context =>
+                {
+                    return Task.FromResult(0);
+                }));
+
+            using (var host = builder.Build())
+            {
+                host.Start();
+
+                using (var socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
+                {
+                    socket.Connect(new IPEndPoint(IPAddress.Loopback, host.GetPort()));
+                    socket.LingerState = new LingerOption(true, 0);
+                }
+
+                // Give it some time for the connection error log message to be logged
+                var filteredWrites = testSink.Writes.Where(write => write.EventId.Id == connectionErrorEventId);
+                for (var i = 0; i < 10; i++)
```

Comment:
I would make the test code itself wait forever (rather than only 500ms), then wrap the whole test in a call to `Task.Run() .TimeoutAfter()` with a large timeout (say 10 seconds), so it fails if it exceeds the timeout for any reason.